Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```diff
@@ -43,8 +44,13 @@ export class LGServer {
   protected readonly documents = new TextDocuments();
   protected readonly pendingValidationRequests = new Map<string, number>();
   protected LGDocuments: LGDocument[] = [];
+  private memoryVariables: Record<string, any> = {};
 
-  constructor(protected readonly connection: IConnection, protected readonly resolver?: FileResolver) {
+  constructor(
+    protected readonly connection: IConnection,
+    protected readonly resolver?: FileResolver,
+    protected readonly memoryResolver?: FileResolver
```

Comment:
First, this memoryResolver don't have to FileResolver, why this is a "FileResolver"??

Reply:
Second, you should also enable the E2E workflow that a set of pre-defined variables through your defined interface here, to prove you interface work.

Reply:
By "focus on interface", i'm not referring to "removing anything besides", the use case of this interface is part of interface design.